tests/sort-requires.js

```diff
@@ -0,0 +1,623 @@
+        { message: "Expected require declaration of 'd' to be placed before 'Z'.", type: 'VariableDeclaration' }
+      ],
+      options: [{ ignoreCase: true }]
+    },
+    // Declaration sort - `ignoreCase` option with `autofixDeclarationSortOrder`
+    {
+      code: `
+        const { G, h } = require('foo');
+        const { e, f } = require('foo');
+        const { C } = require('foo');
+        const B = require('foo');
+        const a = require('foo');
+        const { ...Z } = require('foo');
+        const d = require('foo');
+      `,
+      errors: [
+        { message: "Expected require declaration of 'e' to be placed before 'G'.", type: 'VariableDeclaration' },
+        { message: "Expected require declaration of 'B' to be placed before 'C'.", type: 'VariableDeclaration' },
+        { message: "Expected require declaration of 'a' to be placed before 'B'.", type: 'VariableDeclaration' },
+        { message: "Expected require declaration of 'd' to be placed before 'Z'.", type: 'VariableDeclaration' }
+      ],
+      options: [{ autofixDeclarationSortOrder: true, ignoreCase: true }],
+      output: `
+        const { e, f } = require('foo');
+        const { G, h } = require('foo');
+        const a = require('foo');
+        const B = require('foo');
+        const { C } = require('foo');
+        const d = require('foo');
+        const { ...Z } = require('foo');
+      `
+    },
+    // Declaration sort - `useOldSingleMemberSyntax` option.
+    {
+      code: `
+        const c = require('foo');
+        const { a } = require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'multiple' syntax to be placed before 'single' syntax.",
+          type: 'VariableDeclaration'
+        }
+      ],
+      options: [{ useOldSingleMemberSyntax: true }]
+    },
+    {
+      code: `
+        const c = require('foo');
+        const { ...Z } = require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'multiple' syntax to be placed before 'single' syntax.",
+          type: 'VariableDeclaration'
+        }
+      ],
+      options: [{ useOldSingleMemberSyntax: true }]
+    },
+    // Declaration sort - `useOldSingleMemberSyntax` option with `autofixDeclarationSortOrder`.
+    {
+      code: `
+        const c = require('foo');
+        const { a } = require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'multiple' syntax to be placed before 'single' syntax.",
+          type: 'VariableDeclaration'
+        }
+      ],
+      options: [{ autofixDeclarationSortOrder: true, useOldSingleMemberSyntax: true }],
+      output: `
+        const { a } = require('foo');
+        const c = require('foo');
+      `
+    },
+    {
+      code: `
+        const c = require('foo');
+        const { ...Z } = require('foo');
```

Comment:
Can't we auto fix this to `const Z = require('foo');`? :joy:

Reply:
Ahaha I guess it's out of the scope of this project :D